src/Bindings/ManualBindings.cpp

```diff
@@ -2823,62 +2823,90 @@ class cLuaBlockTracerCallbacks :
 	{
 	}
 
-	virtual bool OnNextBlock(int a_BlockX, int a_BlockY, int a_BlockZ, BLOCKTYPE a_BlockType, NIBBLETYPE a_BlockMeta, eBlockFace a_EntryFace) override
+	virtual bool OnNextBlock(Vector3i a_Block, BLOCKTYPE a_BlockType, NIBBLETYPE a_BlockMeta, eBlockFace a_EntryFace) override
 	{
 		bool res = false;
 		if (!m_Callbacks->CallTableFn(
 			"OnNextBlock",
-			a_BlockX, a_BlockY, a_BlockZ, a_BlockType, a_BlockMeta, a_EntryFace,
+			a_Block, a_BlockType, a_BlockMeta, a_EntryFace,
 			cLuaState::Return, res
 		))
 		{
-			// No such function in the table, skip the callback
-			return false;
+			if (!m_Callbacks->CallTableFn(
```

Comment:
This should be added as part of the if-condition, rather than creating a whole new if statement inside.

Reply:
I tried to change this, I'm not quite sure if I did it the way you thought, but the idea of being not in the same condition is  that the second one only executes if the first doesn't exist.

Reply:
C++ has [short-circuit evaluation](https://en.wikipedia.org/wiki/Short-circuit_evaluation), so in a || b if a == true then b is not evaluated.

Reply:
Thank I didn't know this